CSharpSourceCode/Battle/AI/AgentBehavior/AgentCastingBehavior/AoEAdjacentCastingBehavior.cs

```diff
@@ -0,0 +1,32 @@
+﻿using TaleWorlds.MountAndBlade;
+using TOW_Core.Abilities;
+using TOW_Core.Battle.AI.AgentBehavior.AgentTacticalBehavior;
+using TOW_Core.Battle.AI.Components;
+using TOW_Core.Battle.AI.Decision;
+
+namespace TOW_Core.Battle.AI.AgentBehavior.AgentCastingBehavior
+{
+    public class AoEAdjacentCastingBehavior : AbstractAgentCastingBehavior
+    {
+        public AoEAdjacentCastingBehavior(Agent agent, AbilityTemplate template, int abilityIndex) : base(agent, template, abilityIndex)
+        {
+            TacticalBehavior = new AdjacentAoETacticalBehavior(agent, agent.GetComponent<WizardAIComponent>(), this);
+        }
+
+        public override void Execute()
+        {
+            var castingPosition = ((AdjacentAoETacticalBehavior) TacticalBehavior)?.CastingPosition;
+            if (castingPosition.HasValue && Agent.Position.AsVec2.Distance(castingPosition.Value.AsVec2) > 5) return;
```

Comment:
so agent won't cast if distance to target is more than 5?

Reply:
Kind of. In this case the distance it is from the "casting position" as there is no explicit target. This is specifically for AoE spells that have effects adjacent to the caster. The Wind spell "tactical casting" has a similar behavior but directs the agent to a flank instead.

 Other spells can still be cast from further away.